### PR TITLE
ADD FLY_CDY BOARD

### PR DESCRIPTION
**Pull request by FLYmaker**
### Requirements

* Filling out this template is required. Pull Requests without a clear description may be closed at the maintainers' discretion.

### Description
ADD FLY_CDY BOARD, it brings together heating rod wire, thermistor wire, controllable fan, uncontrollable fan wire, and extruder wire, which is convenient for everyone to connect. And use a more convenient typec data line interface
IMG_20200915_212533
[FLY_CDY SCH.pdf](https://github.com/MarlinFirmware/Marlin/files/5470429/FLY_CDY.SCH.pdf)

And fix the problem that the mini board cannot be compiled.
Add FLYF407ZG board, onboard SPI thermocouple configuration

### CODE REVIEW

File: Marlin/src/pins/lpc1769/pins_FLY_CDY.h [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+/**
+ * Smoothieboard pin assignments
+ */
+
+#ifndef MCU_LPC1769
```
Comment (sjasonsmith) [inline Marlin/src/pins/lpc1769/pins_FLY_CDY.h]:
```suggestion
#if NOT_TARGET(MCU_LPC1769)
```

The `NOT_TARGET` macros avoids problems that can occur during Marlin's dependency detection phase.

File: Marlin/src/core/boards.h [hunk 2]
```diff
@@ -308,18 +308,21 @@
...
-#define BOARD_CCROBOT_MEEB_3DP        4031  // ccrobot-online.com MEEB_3DP (STM32F103RC)
-#define BOARD_CHITU3D_V5              4032  // Chitu3D TronXY X5SA V5 Board
-#define BOARD_CHITU3D_V6              4033  // Chitu3D TronXY X5SA V5 Board
-#define BOARD_CREALITY_V4             4034  // Creality v4.x (STM32F103RE)
-#define BOARD_CREALITY_V427           4035  // Creality v4.2.7 (STM32F103RE)
-#define BOARD_TRIGORILLA_PRO          4036  // Trigorilla Pro (STM32F103ZET6)
+#define BOARD_BTT_SKR_MINI_MZ_V1_0    4025  // BigTreeTech SKR Mini MZ V1.0 (STM32F103RC)
+#define BOARD_BTT_SKR_E3_DIP          4026  // BigTreeTech SKR E3 DIP V1.0 (STM32F103RC / STM32F103RE)
+#define BOARD_JGAURORA_A5S_A1         4027  // JGAurora A5S A1 (STM32F103ZET6)
+#define BOARD_FYSETC_AIO_II           4028  // FYSETC AIO_II
```
Comment (sjasonsmith) [inline Marlin/src/core/boards.h]: Too many lines show up edited here. You probably need to merge or rebase with the latest state of bugfix-2.0.x to resolve this cleanly.

File: Marlin/src/pins/lpc1769/pins_FLY_CDY.h [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+#define BEEPER_PIN                          P2_07
+#define LCD_PINS_RS                         P2_10
+#define LCD_PINS_ENABLE                     P0_22
+#define LCD_PINS_D4                         P1_19
+#define LCD_PINS_D5                         P2_08
+#define LCD_PINS_D6                         P1_30
+#define LCD_PINS_D7                         P1_31
+#define BTN_EN1                             P0_00
+#define BTN_EN2                             P0_01
+#define BTN_ENC                             P0_28
```
Comment (sjasonsmith) [inline Marlin/src/pins/lpc1769/pins_FLY_CDY.h]: What display is this for?

File: Marlin/src/pins/lpc1769/pins_FLY_CDY.h [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+ #if SD_CONNECTION_IS(LCD)
+
+  #define SCK_PIN                           P0_15
+  #define MISO_PIN                          P0_17
+  #define MOSI_PIN                          P0_18
+  #define SS_PIN                            P0_16
+  #define SD_DETECT_PIN                     P2_06
+
+#endif
```
Comment (sjasonsmith) [inline Marlin/src/pins/lpc1769/pins_FLY_CDY.h]: There are some issues with this block of defines. Is this what you meant?

```suggestion
#if SD_CONNECTION_IS(ONBOARD)
  #define SS_PIN                          P0_06
  #define SCK_PIN                         P0_07
  #define MISO_PIN                        P0_08
  #define MOSI_PIN                        P0_09
  #define SD_DETECT_PIN                   P0_05
#elif SD_CONNECTION_IS(LCD)
  #define SCK_PIN                         P0_15
  #define MISO_PIN                        P0_17
  #define MOSI_PIN                        P0_18
  #define SS_PIN                          P0_16
  #define SD_DETECT_PIN                   P2_06
#endif
```

File: Marlin/src/pins/pins.h [hunk 5]
```diff
@@ -423,6 +423,10 @@
   #include "lpc1769/pins_MKS_SGEN_L_V2.h"       // LPC1769                                env:LPC1769
 #elif MB(BTT_SKR_E3_TURBO)
   #include "lpc1769/pins_BTT_SKR_E3_TURBO.h"    // LPC1769                                env:LPC1769
+#elif MB(FLY_CDY)
+  #include "lpc1769/pins_FLY_CDY.h"             // LPC1769                                env:LPC1769
+#elif MB(FLSUN_HISPEED)
+  #include "stm32f1/pins_FLSUN_HISPEED.h"       // STM32F1                                env:flsun_hispeed
```
Comment (sjasonsmith) [inline Marlin/src/pins/pins.h]: This should not be here

File: Marlin/src/pins/pins.h [hunk 6]
```diff
@@ -569,6 +573,10 @@
   #include "stm32f1/pins_CREALITY_V427.h"       // STM32F1                                env:STM32F103RET6_creality
 #elif MB(TRIGORILLA_PRO)
   #include "stm32f1/pins_TRIGORILLA_PRO.h"      // STM32F1                                env:trigorilla_pro
+#elif MB(FLY_MINI)
+  #include "stm32f1/pins_FLY_MINI.h"            // STM32F1                                env:FLY_MINI
+#elif MB(FLSUN_HISPEED)
+  #include "stm32f1/pins_FLSUN_HISPEED.h"       // STM32F1                                env:flsun_hispeed
```
Comment (sjasonsmith) [inline Marlin/src/pins/pins.h]: This should not be showing up in your PR. I'm guessing that rebasing/merging will make it go away,

File: Marlin/src/pins/stm32f1/pins_FLY_MINI.h [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+#if NOT_TARGET(__STM32F1__)
+  #error "Oops! Select an STM32F1 board in 'Tools > Board.'"
+#endif
+
+#define BOARD_INFO_NAME   "FLY_MINI"
```
Comment (sjasonsmith) [inline Marlin/src/pins/stm32f1/pins_FLY_MINI.h]: There is something really weird with this PR. FLY_MINI is already in bugfix, so why do these look like new files?
I am guessing your changes were made on an old version of bugfix-2.0.x, and you have some unresolved merge issues.
https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.0.x/Marlin/src/pins/stm32f1/pins_FLY_MINI.h

**Review by sjasonsmith — CHANGES_REQUESTED** [review]

File: Marlin/src/pins/lpc1769/pins_FLY_CDY.h (hunk 3 above)
Comment (FLYmaker) [reply Marlin/src/pins/lpc1769/pins_FLY_CDY.h]: Here is the 12864LCD screen

File: Marlin/src/pins/lpc1769/pins_FLY_CDY.h (hunk 4 above)
Comment (FLYmaker) [reply Marlin/src/pins/lpc1769/pins_FLY_CDY.h]: Yes. . I seem to make a mistake

File: Marlin/src/pins/stm32f1/pins_FLY_MINI.h (hunk 7 above)
Comment (FLYmaker) [reply Marlin/src/pins/stm32f1/pins_FLY_MINI.h]: Yes, the configuration of MINI has not been modified before I submit

**Comment by sjasonsmith** [comment]
@FLYmaker when posting future PRs please make a new branch used only for the PR. This was created from your `bugfix-2.0.x` branch, and had a mixture of three unrelated changes, which had to be cleaned up before this could be completed.

In the merge to clean up this PR, I left out some changes related to your FLY407 board from commits 383c0de22a7cdc3d1977f2cac8bd7ad360ea2c0b  and 5724821f0b7f8586a7a4a98baa166d26387b943e. If those changes are needed please post them in a new PR.

**Comment by sjasonsmith** [comment]
I did not even see this comment in the PR description, and I did not include them when I cleaned up the FLY_CDY files.

```
And fix the problem that the mini board cannot be compiled.
Add FLYF407ZG board, onboard SPI thermocouple configuration
```

If these changes are still needed, please post them in a separate PR.

**Comment by thinkyhead** [comment]
I now have some FLY boards in hand for testing. They are very nice looking and I look forward to testing them out.

**Pull request merged**